src/nvim/state.c

```diff
@@ -121,6 +121,7 @@ void state_handle_k_event(void)
     if (multiqueue_empty(main_loop.events)) {
       // don't breakcheck before return, caller should return to main-loop
       // and handle input already.
+      multiqueue_process_events(deferred_autocmd_events);
```

Comment:
@justinmk let me know if you were thinking of a different spot to process events in `state_handle_k_event`.

Reply:
I don't think this makes sense. `deferred_autocmd_events` is already a child queue of `main_loop.events`, so `deferred_autocmd_events` is already empty when `main_loop.events` is empty.

Reply:
any reason not to do it just before the `os_breakcheck` ?

side-note, now I'm wondering if `aucmd_defer` should just add directly to `main_loop.events`, maybe a separate queue isn't needed? just something to keep in mind.

Reply:
I think there is another problem. `K_EVENT` is only processed when there is no pending input, which means when executing a macro `MarkSet` autocommands are postponed after the end of the macro.

Reply:
we can deal with that by processing the queue in multiple places, right?

alternatively, queuing into `main_loop.events` would also address that?

Reply:
Both queues work in my testing. (See both `test/functional/autocmd/markset_spec.lua` and the video above.)

Current approach keeps the `deferred_autocmd_events` queue but let me know if I should replace it with `main_loop.events`.

Reply:
Note: the feature works event without any explicit queue processing: https://github.com/neovim/neovim/pull/35793#discussion_r2366829750